interview-questions.md

```diff
@@ -5,64 +5,88 @@ First, without external resources. Challenge yourself to answer from memory as i
 
-  Researched answer:
+  the difference: Element 0 is at index 0. "Index" describes the storage location. "Element" describes the contents of a location. "Value" describes the content of an element.
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: an iteration is a code instructions that keep going until its told to stop or gets the desired result
 
-  Researched answer:
+  Researched answer: Iteration is the process of executing a block of code over and over again until a condition is met.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Proper pair programming techniques, means someone is always a driver or a navigator. This involves the navigator watching for errors in the code. The navigator should also, have related topics open to the problem at hand. This helps the driver and navigator accomplish their goal together as a team. The driver should be writing code and assessing the problem at hand as they go. Drivers should also follow the navigators instruction. As well as share their screen, so navigators can watch for errors and assist with the code creation and debugging.Teams of drivers and navigators should switch off. This assists in problem solving, brings a fresh look at the code when you may be stuggling,and helps with possible unseen errors. My personal pair programming technique is knowing when to say, your unsure how to start or get the desired result you are looking for. allowing you to gain understanding of your partners ideas on a possible solution, their knowledge which you may not have, and create a solution together.Which you may have not come up with on your own. Rather than one person coding and the other just watching for errors and possibly never coming to a solution.
 
-  Researched answer:
+  Researched answer: Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator,reviews each line of code as it is typed in. The two programmers switch roles frequently. While reviewing, the observer also considers the "strategic" direction of the work, coming up with ideas for improvements and likely future problems to address. This is intended to free the driver to focus all of their attention on the "tactical" aspects of completing the current task, using the observer as a safety net and guide. a techniqe from the slide show on github from the pair programming introduction says as followed "Be strong enough to put yourself in the vulnerable position of asking for help. Ask your partner to slow down and talk through the process. Ask for clarification on code you dont understand and take notes of confusing point to review later."
```

Comment:
Please make sure to form researched answers in your own words.

Reply:
doing so now i'll push my code before class today